### DISCUSSION TITLE
Aria2设置成功却无法使用。发送到Aria2和离线下载均无法使用。

### DISCUSSION THREAD

**Discussion by wuhuqifeiQWER** [AlistGo/alist#2845] [Q&A] [answered]
alist提示aria2设置成功
![image](https://user-images.githubusercontent.com/106081176/210058510-fb39132a-499e-4c89-ab21-c491805db75d.png)
下载任务发送到aria2成功
![image](https://user-images.githubusercontent.com/106081176/210058635-25f9b7a4-1110-4de1-a288-036203e8c268.png)
而实际上aria2下载失败
![image](https://user-images.githubusercontent.com/106081176/210058684-bce686a3-d028-4000-8520-0b49430d1d97.png)
aria2程序报错内容如下
![image](https://user-images.githubusercontent.com/106081176/210064933-c9be5660-65ff-4c3d-b75d-51d7aa2f84ae.png)

离线下载的Aria2可以正常使用。

个人目前感觉是aria2秘钥没有填写？
但是我在AriaNG里设置了aria2 RPC 秘钥，并且尝试填写进了alist的设置（管理界面的“设置”--“其他”和主页的“本地设置”--“Aria2 RPC 秘钥”都填写了），但情况并没有改变，aria2秘钥和aria2 RPC秘钥不是同一个东西吗？如果不同，aria2秘钥如何获取？我是Windows本地使用，是直接使用cmd运行aria2，并未返回类似秘钥的值。
![image](https://user-images.githubusercontent.com/106081176/210059464-80a9196c-0d76-445e-8bb4-f3ec14413969.png)

求大佬们指教。

**Comment by 121909** [comment]
你这个可能是发送到aria2的地址错误了
![image](https://user-images.githubusercontent.com/63403797/210365513-8841012d-a3d9-4a09-b217-3351da7f50f9.png)
如果你的alist是在远程运行的话，应该是你的网址而不是localhost

**Reply by wuhuqifeiQWER** [reply]
不，不是的，alist和aria2都是在本地运行的。离线下载功能正常，而alist下载文件时发送到aria2无法正常下载。

**Comment by NovaShen0724** [comment, marked as answer]
可以尝试ping localhost并继续排查

**Reply by wuhuqifeiQWER** [reply]
![image](https://user-images.githubusercontent.com/106081176/210382263-5cd564fa-681c-4ee2-b999-fe42f76c8d0e.png)
并未发现问题

**Reply by NovaShen0724** [reply]
尝试提供下载时alist log并尝试使用远程设备下载

**Reply by NovaShen0724** [reply]
我觉得可能是回环的问题，你可以试试127.0.0.1去访问这样
参考https://github.com/aria2/aria2/issues/1117

**Reply by wuhuqifeiQWER** [reply]
![image](https://user-images.githubusercontent.com/106081176/210383843-35c62ec3-1974-47fa-9c4b-03c9ec02680a.png)
aria2设置成功log，下载文件选择“发送到aria2”后，未返回log

**Reply by wuhuqifeiQWER** [reply]
![image](https://user-images.githubusercontent.com/106081176/210383929-363d14b6-9774-4837-b41a-ff2baac77126.png)
aria2依然下载失败

**Reply by NovaShen0724** [reply]
尝试访问127.0.0.1下载如何？

**Reply by wuhuqifeiQWER** [reply]
![image](https://user-images.githubusercontent.com/106081176/210384448-59cac827-60ae-4a46-9551-b1330d5b1c9f.png)
成功，问题已解决，感谢解答。
顺便问一下，这种是不能同时出现2个localhost？

**Reply by NovaShen0724** [reply]
根据aria2报错来看，似乎是aria2将localhost当成域名向dns发送请求导致的错误，原因未知，可以Google一下原因

**Comment by rsykvjhp** [comment]
我找到原因了，把aria2高级设置中的异步dns改成否就可以了，我用的是AriaNg
[![异步dns惹的祸.png](https://p.inari.site/usr/805/65f2f67775c67.png)](https://p.inari.site/usr/805/65f2f67775c67.png)
